habitat/core/env.py

```diff
@@ -20,6 +20,7 @@
 from habitat.datasets import make_dataset
 from habitat.sims import make_sim
 from habitat.tasks import make_task
+from habitat_sim.utils import profiling_utils
```

Comment:
Thus far we have kept habitat-sim as a soft dependency for habitat-lab.  This is currently crucial for the habitat-pyrobot bridge as it lets people not have to worry about figuring out how to compile habitat-sim on the robot platform (and that platform may not have the RAM requirements to compile the python bindings in some cases).

However, I get the desire to share code between habitat-lab and habitat-sim.  Maybe we need to consider having habitat_utils be its own thing?  We could keep that in the habitat-sim repo (to not have a 3rd to manage) and then the CI that does nightly Conda-builds could also upload that to Conda and pip for us since that package would be pure python and platform agnostic?

CC @mathfac

Reply:
Yeah, I tried to bring attention to this a while back. See Slack discussion: https://cvmlp.slack.com/archives/CFN5TAUSD/p1597880436485300 .

Reply:
From that thread, only Dhruv really weighed in:
>Yeah, I think hab-sim being a hard dependency for hab-lab is fine.
But I understand we can't break the pyrobot bridge. I don't know if what I'm doing here is breaking that.

>Maybe we need to consider having habitat_utils be its own thing? We could keep that in the habitat-sim repo (to not have a 3rd to manage) 
This sounds reasonable and is one of the approaches the infra team has previously discussed. So this would be a different bindings binary, set up alongside habitat_sim, but both live in the habitat-sim repo.

What do you want to see for this PR? Change `import habitat_sim` to `import habitat_utils`? Or is that a refactor we can tackle later?

Reply:
Yeah, I saw that message and it didn't click that the pyrobot bridge would break by doing that until now.  I am not entirely certain what is the right call for this.

Reply:
One option is to punt on this whole problem for now and just move `profiling_utils` back into Hab-lab. I kinda wanted it available to Hab-sim python code, but that's not super important.

Reply:
Yeah, for now it is an easier problem to install habitat-lab to get the profiling tools than it is to make habitat-sim a hard dependency due to pyrobot, but this does bring up a good point of where do we keep common utils.  There haven't be any significant ones yet, so the little ones have just gotten duplicated (see the `quat_rotate_vector` function)

Reply:
Maybe we want to create a unified `habitat_utils` module (for python only) in habitat-sim repo outside of the other source and place all utils which are not C++ dependent here. Then folks could clone habitat-sim but would not need to build it to access these.

Reply:
Ok. For the 0.1.6 release, I'm going to leave `profiling_utils` in habitat-sim's utils folder.

After code freeze, I'll investigate setting up `habitat_utils`. @erikwijmans , how do I go about testing the pyrobot bridge or verifying it's still working? Is there CI or a unit test?

Reply:
Can't we just have a dummy class with no op functions if it fails to import?

Reply:
After all this time, we still haven't found a good way to do this `habitat_utils`, although we're still discussing.

For now, I'll take @Skylion007 's suggested approach of importing with a no-op fallback when the import fails. In other words, habitat_sim is still optional when using habitat-lab (but you don't get profiling if you don't have habitat-sim installed, which is fine).